src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -725,6 +727,26 @@ public function appendToFile($filename, $content)
         }
     }
 
+    /**
+     * Watches a file or directory for any changes, and calls $callback when any changes are detected.
+     *
+     * @param mixed    $path     The path to watch for changes. Can be a path to a file or directory, iterator or array with paths
+     * @param callable $callback The callback to execute when a change is detected
+     * @param float    $timeout  The time in milliseconds to wait between checking for changes (defaults to 1000 when inotify is not available)
```

Comment:
Sorry I wasn't clear enough in my suggestion here: I meant an idle timeout - ie if nothing happens for that duration, we should quit the while loop in implementations.
But as long as something happens during the interval, we should continue looping.
By default or `null` should mean : forever (as is does now already - configuring the pause timeout doesn't provide any useful behavior from the outside)

Reply:
Can you clarify how is such timeout useful? As a user that looks highly undesired for watcher to terminate even after one hour of no changes. I wish watcher to run until I terminate it. More useful would be original interpretation of timeout.